eclipse-collections/src/main/java/org/eclipse/collections/impl/block/factory/Comparators.java

```diff
@@ -553,4 +563,30 @@ public int compare(T one, T two)
             return ((Comparable<T>) one).compareTo(two);
         }
     }
+
+    private static final class FromPredicateSerializableComparator<T> implements SerializableComparator<T>
+    {
+        private static final long serialVersionUID = 1L;
+
+        private final BiPredicate<? super T, ? super T> isBeforePredicate;
+
+        private FromPredicateSerializableComparator(BiPredicate<? super T, ? super T> isBeforePredicate)
+        {
+            this.isBeforePredicate = isBeforePredicate;
+        }
+
+        @Override
+        public int compare(T o1, T o2)
+        {
+            if (this.isBeforePredicate.test(o1, o2))
+            {
+                return -1;
+            }
+            if (this.isBeforePredicate.test(o2, o1))
+            {
+                return 1;
+            }
+            return 0;
+        }
+    }
```

Comment:
Maybe I should called it isLessThan instead of isBefore?

Reply:
@motlin isLessThanPredicate might make more sense here. I usually think of isBefore in terms of time.